test/clj_sqs_extended/core_test.clj

```diff
@@ -111,6 +114,61 @@
           (is (contains? stats :stopped-at))))
       (close! handler-chan))))
 
+(deftest handle-queue-terminates-after-restart-count-exceeded
+  (testing "handle-queue terminates when the restart-count exceeds the limit"
+    (let [handler-chan (chan)]
+      (fixtures/with-test-standard-queue
+        (with-redefs-fn {#'sqs/receive-message
+                         (fn [_ _ _] (HttpTimeoutException.
+                                       "Something seems to be wrong with the network."))}
+          #(let [stop-fn (fixtures/with-handle-queue-queue-opts-standard-no-autostop
+                           handler-chan
+                           {:restart-limit         3
+                            :restart-delay-seconds 1})]
+             (Thread/sleep 3500)
+             (let [stats (stop-fn)]
+               (is (= (:restart-count stats) 3))))))
+      (close! handler-chan))))
+
+(deftest handle-queue-restarts-after-recoverable-error-occured
+  (testing "handle-queue restarts properly and continues running upon recoverable error"
+    (let [handler-chan (chan)]
+      (fixtures/with-test-standard-queue
+        (let [stop-fn (fixtures/with-handle-queue-queue-opts-standard-no-autostop
+                        handler-chan
+                        {:restart-delay-seconds 1})]
+          ;; Cause some trouble in receive-message ...
+          (trouble/activate-troublemaker :receive-message)
+          (Thread/sleep 1500)
+
+          ;; All good again, back to work!
+          (trouble/deactivate-troublemaker :receive-message)
+          (Thread/sleep 1500)
+
+          ;; Verify that the queue is still handled properly ...
+          (is (string? (sqs-ext/send-message @fixtures/test-sqs-ext-client
+                                             fixtures/test-standard-queue-name
+                                             (first test-messages-basic))))
+          (is (= (first test-messages-basic) (:body (<!! handler-chan))))
+          (let [stats (stop-fn)]
+            ;; ... but receiving was actually restarted
+            (is (>= (:restart-count stats) 0)))))
```

Comment:
this can be accomplished by using the `circleci/bond` library and keep a `bond/with-spy` on one of the fns involved with restarting the loop. that's recommended so that we don't need to keep track of this in production code. 

we need to focus this codebase to do the bare essentials and other logic should be minimized by outsourcing to other libraries. i.e. I don't want to end up maintaining this restart-count logic.

Reply:
Roger that, I'll have a look at it and rework it as you suggested.